docs/architecture/adr-049-state-sync-hooks.md

```diff
@@ -148,7 +154,7 @@ As a result of this implementation, we are able to create snapshots of binary ch
 
 ### Backwards Compatibility
 
-This ADR introduces new proto message types, add field for extension snapshotters to snapshot `Manager`, add new function signatures `SnapshotFormat()`, `SupportedFormats()` to `Snapshotter` interface and add new `NamedSnapshotter` interface, so this is not backwards compatible.
+This ADR introduces new proto message types, add field for extension snapshotters to snapshot `Manager`, add new function signatures `SnapshotFormat()`, `SupportedFormats()` to `Snapshotter` interface and add new `ExtensionSnapshotter` interface, so this is not backwards compatible.
```

Comment:
It might be worth mentioning that the wire format of the snapshot is backward compatible when there's no `ExtensionSnapshotter` registered.

Reply:
sure, I think this is significant for application chains that don't have additional state to take snapshots.
So for application chains of this kind, this ADR should be completely backwards compatible?

Reply:
Yes, let's add some wording that the stream is backwards-compatible if no extension is registered.